lib/workers/repository/model/commit-message.ts

```diff
@@ -44,6 +46,15 @@ export class CommitMessage {
     return [prefix, message].join(' ').trim();
   }
 
+  public static formatCasing(commitMessage: string): string {
```

Comment:
I still think we should do it before we generate the full message.

Reply:
```js
if (upgrade.semanticCommitCasing) {
      // We only need to lowercase the first word of the commitMessage
      const splitMessage = upgrade.commitMessage.split('\n');
      splitMessage[0] = CommitMessage.formatCasing(splitMessage[0]);
      upgrade.commitMessage = splitMessage.join('\n');
    }
    ```
We only lowercase the first word incase `upgrade.semanticCommitCasing` is true, so is it necessary to change the before this.

Reply:
Can't it be done somewhere around https://github.com/renovatebot/renovate/blob/0e2f83ae6d7a5297f9ee4d9990245b063e3942ae/lib/workers/repository/model/commit-message.ts#L44 and avoid the regex?

Reply:
We have 4 different types of commit-messages so using regex is the better approach I think.

We can put the functionality in `toString` method and use it in the `generate.ts` file. There is no problem I could find  with that except the case that it has been used in two other files and I am not sure if we want to change those commit-messages or not.
The 2 files are:-
https://github.com/renovatebot/renovate/blob/8c588ddbfc4a6deba1521401573422485163db67/lib/workers/repository/onboarding/branch/create.ts#L41

https://github.com/renovatebot/renovate/blob/8c588ddbfc4a6deba1521401573422485163db67/lib/workers/repository/onboarding/branch/rebase.ts#L58

Reply:
@RahulGautamSingh @viceice Could you please read the above comment, and guide as to what changes need to be made.

Reply:
We don't want to use regex for this. Can you find a way?

Reply:
On it.